docs/graph-homomorphism-as-natural-transformation.md

```diff
@@ -58,6 +68,10 @@ We then equate this to a graph by stating that:
  * A graph consists of vertexes V.
  * A graph consists of edges E, which are defined as products of vertexes V x V.
 
+[Paul: Note: the edge set E is a subset of V x V, not necessarily all of V x V.  Indeed, the edge set is all of V x V if and only if G is the complete directed graph on the vertex set V.
+
+Actually, if one wants to consider directed multigraphs (multiple edges between the same pair of vertices are allowed) then you can't really even think of E as a subset of V x V.  Instead you think of E as just an abstract set equipped with a map E -> V x V which sends every edge e to its source and target.  This fits well with the functorial definition of a graph.]
```

Comment:
With a multigraph wouldn't it be the case that the mapping was (potentially) surjective but not injective?

Is it useful to have a multigraph where you have multiple identical edges? I think the source and destination of the edge need not be the only thing on it (e.g. it could be a weighted graph). If a graph exists which labels each edge with additional information and then I define a mapping which drops the edge label then wouldn't that map a multigraph to a regular graph?

I guess what I'm saying is how can a directed multigraph, with no additional information on the edges, be defined where two edges exist between the same pair of vertices and yet those edges are not the same edge?

Since you've stated that E may not be a subset of V x V then I guess that sets are not required to contain unique values. This is surprising because when programming the one quality of a set is uniqueness of elements.